File: pgpainless-core/src/main/java/org/pgpainless/signature/subpackets/SignatureSubpacketCallback.java
// SPDX-FileCopyrightText: 2021 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package org.pgpainless.signature.subpackets;

public interface SignatureSubpacketCallback<S extends BaseSignatureSubpackets> {

    /**
     * Callback method that can be used to modify the hashed subpackets of a signature.
     *
     * @param hashedSubpackets hashed subpackets
     */
    default void modifyHashedSubpackets(S hashedSubpackets) {

    }

    /**
     * Callback method that can be used to modify the unhashed subpackets of a signature.
     *
     * @param unhashedSubpackets unhashed subpackets
     */
    default void modifyUnhashedSubpackets(S unhashedSubpackets) {

    }
}

File: pgpainless-core/src/main/kotlin/org/pgpainless/signature/subpackets/SignatureSubpacketCallback.kt
// SPDX-FileCopyrightText: 2023 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package org.pgpainless.signature.subpackets

interface SignatureSubpacketCallback<S : BaseSignatureSubpackets> {

    /**
     * Callback method that can be used to modify the hashed subpackets of a signature.
     *
     * @param hashedSubpackets hashed subpackets
     */
    fun modifyHashedSubpackets(hashedSubpackets: S) {
        // Empty default implementation to allow for cleaner overriding
    }

    /**
     * Callback method that can be used to modify the unhashed subpackets of a signature.
     *
     * @param unhashedSubpackets unhashed subpackets
     */
    fun modifyUnhashedSubpackets(unhashedSubpackets: S) {
        // Empty default implementation to allow for cleaner overriding
    }
}
